installinator/src/dispatch.rs

```diff
@@ -203,6 +204,13 @@ impl InstallOpts {
                         fetch_artifact(&cx, &host_phase_2_id, discovery, log)
                             .await?;
 
+                    check_downloaded_artifact_hash(
+                        "host phase 2",
+                        host_phase_2_artifact.artifact.clone(),
+                        host_phase_2_id.hash,
+                    )
+                    .await?;
```

Comment:
Do we want to retry the operation if it fails? (If the answer is "not for now, maybe in the future", could you add a comment about it here?)

Reply:
I think not, maybe ever? We're fetching this data from wicketd (in memory) over TCP to installinator (in memory), so I think the by-far-most-likely reason for this to fail is if wicketd sends the wrong artifact, which I wouldn't expect to be resolved by a retry (and we would definitely want to know immediately if it happened). I'll add a comment though :+1:.

Reply:
Added explanatory comments in [8ea3902](https://github.com/oxidecomputer/omicron/pull/3455/commits/8ea3902187af7918fd2e469becca7e7dcb9d604f)